File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_154/go/solution.go
package cycpattern_check_test

import (
    "testing"
    "fmt"
    "strings"
)

// You are given 2 words. You need to return true if the second word or any of its rotations is a substring in the first word
// >>> cycpattern_check("abcd", "abd")
// false
// >>> cycpattern_check("hello", "ell")
// true
// >>> cycpattern_check("whassup", "psus")
// false
// >>> cycpattern_check("abab", "baa")
// true
// >>> cycpattern_check("efef", "eeff")
// false
// >>> cycpattern_check("himenss", "simen")
// true
func cycpattern_check(a string, b string) bool {
    doubled := b + b
    patternLen := len(b)
    for i := 0; i < patternLen; i++ {
        if strings.Contains(a, doubled[i:i+patternLen]) {
            return true
        }
    }
    return false
}

File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_154/python/solution.py
def cycpattern_check(text: str, pattern: str) -> bool:
    """Check if pattern or any rotation of it is a substring of text.
    
    cycpattern_check("abcd","abd") => False
    cycpattern_check("hello","ell") => True
    cycpattern_check("whassup","psus") => False
    cycpattern_check("abab","baa") => True
    cycpattern_check("efef","eeff") => False
    cycpattern_check("himenss","simen") => True
    """
    return any(pattern[i:] + pattern[:i] in text for i in range(len(pattern)))
